src/libthread/daemonize.c

```diff
@@ -106,7 +106,7 @@ _threadsetupdaemonize(void)
 	 * Put it in its own process group so that we don't get a SIGHUP
 	 * when the parent exits.
 	 */
-	setpgrp();
+	setpgid(0, 0);
```

Comment:
Anthony, forgive me for my ignorance, but I've been wondering (since the intial PR) why `rfork(RFNOTEG)` isn't used instead.